Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp

```diff
@@ -2549,6 +2549,170 @@ JS_DEFINE_NATIVE_FUNCTION(@class_name@::to_json)
 )~~~");
 }
 
+static void generate_named_properties_object_declarations(IDL::Interface const& interface, StringBuilder& builder)
+{
+    SourceGenerator generator { builder };
+
+    generator.set("named_properties_class", DeprecatedString::formatted("{}Properties", interface.name));
+
+    generator.append(R"~~~(
+class @named_properties_class@ : public JS::Object {
```

Comment:
I'm not 100% sold on doing this here in the Bindings Generator. There's so much raw text it might make sense to just implement this in ... HTML/ ? Bindings/ ?